src/screens/EtherspotDeploymentInterjection/EtherspotDeploymentInterjection.js

```diff
@@ -55,16 +77,46 @@ import { firebaseRemoteConfig } from 'services/firebase';
  * Interjection screen used when Etherspot smart wallet is not yet deployed on given chain.
  */
 function EtherspotDeploymentInterjection() {
-  const { t } = useTranslationWithPrefix('etherspot.deploymentInterjection');
+  const { t, tRoot } = useTranslationWithPrefix('etherspot.deploymentInterjection');
+  const dispatch = useDispatch();
   const navigation = useNavigation();
+  const fiatCurrency = useFiatCurrency();
   const [interjectionPrismicContent, setInterjectionPrismicContent] = React.useState({});
   const [introductionText, setIntroductionText] = React.useState('');
   const [isPrismicContentFetched, setIsPrismicContentFetched] = React.useState(false);
   const [errorMessage, setErrorMessage] = React.useState('');
+  const [showDeploymentFeeInToken, setShowDeploymentFeeInToken] = React.useState(false);
+  const gasInfo = useRootSelector((root) => root.history.gasInfo);
 
   const chain: Chain = navigation.getParam('chain') ?? CHAIN.ETHEREUM;
+
+  const chainRates = useChainRates(chain);
+
+  React.useEffect(() => {
+    dispatch(fetchGasInfoAction(chain));
+  }, [dispatch, chain]);
+
   const prismicInterjectionDocumentId = firebaseRemoteConfig.getString(REMOTE_CONFIG.PRISMIC_INTERJECTION_DOCUMENT_ID);
 
+  const deploymentFee = React.useMemo(() => {
```

Comment:
Consider extracting `deploymentFee` (in native asset) calculation into hook `useDeployementFee` containing lines from 102 (start) to 110 (deploymentFeeBN). Other operations, like fiat calculation and formatting could be doing on every render

Reply:
I will leave as this move would require either: more checks to be added before calculating into display values, some repeated code to import same selectors and etc. I find this piece being more pragmatic.